_posts/2019-08-26-how-to-use-gruntwork-infrastructure-as-code-library.adoc

```diff
@@ -1399,6 +1409,363 @@ module "vpc" {
+
+In all cases, you'll need to ensure that your workspaces stay in sync with your Terragrunt configuration. Each time you add a new module in Terragrunt, you'll need a corresponding workpace. Furthermore, if you rotate your AWS API keys, you'll need to update them within each workspace. For that reason, the final option above is recommended.
+
+===== Setting variables
+
+In typical Terragrunt usage, variables are passed to Terraform using the https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#inputs[`inputs`] attribute. Anything defined in the inputs will be passed to Terraform using https://www.terraform.io/docs/configuration/variables.html#environment-variables:[the environment variable mechanism]. However, this mechanism is not supported by TFC. Instead, you can generate a `*.auto.tfvars` file containing the inputs.
+
+IMPORTANT: The generated `terragrunt.auto.tfvars` file will be present on local disk. Depending upon your configuration, this file may contain sensitive data. Do not commit this file to version control. We recommend adding the `*.auto.tfvars` to `.gitignore`.
+
+The following code shows how to generate a `terragrunt.auto.tfvars` file.
+
+.infrastructure-live/dev/us-east-1/sqs/terragrunt.hcl
+[source,hcl]
+----
+terraform {
+  source = "git::ssh://git@github.com/gruntwork-io/package-messaging//modules/sqs?ref=v0.3.2"
+}
+
+include {
+  path = find_in_parent_folders()
+}
+
+locals {
+  common_vars = read_terragrunt_config(find_in_parent_folders("common.hcl"))
+}
+
+generate "tfvars" {
+  path      = "terragrunt.auto.tfvars"
+  if_exists = "overwrite"
+  disable_signature = true
+  contents = <<-EOF
+name = "${local.common_vars.locals.name}"
+EOF
+}
+----
+
+The configuration has a few sections:
+
+1. The `terraform` block at the top uses the Gruntwork `sqs` module from https://github.com/gruntwork-io/package-messaging/[`package-messaging`].
+1. The `include` block includes the configuration from the parent directories. This is how the remote `backend` block from the root `terragrunt.hcl` is included.
+1. The `locals` block reads the values from `common.hcl` in the root of the hierarchy, making them available for local reference.
+1. Finally, the `generate` block creates a file called `terragrunt.auto.tfvars`. Like the `backend.tf` file, this file will be generated alongside the rest of the `*.tf` files that Terragrunt downloads from the `sqs` module, making those inputs available for TFC to read when running `terraform` commands in the remote executor.
+
+Any of the inputs needed by the module must be included in the generated tfvars file. In the configuration above, only
+the `name` variable is specified. Most modules will need more configuration.
+
+
+===== Summarizing Terragrunt and TFC/TFE compatibility
```

Comment:
Add one section just before this where you run `terragrunt apply` and show in the CLI output that the `plan` and `apply` actually happen remotely, in TFC, which stores the state and maintains a log of all these runs.

Reply:
ed6c196